src/SnarkBasedFeeOracle.sol

```diff
@@ -41,47 +40,72 @@ contract SnarkBasedFeeOracle is RvVerifier, Ownable {
     }
 
-    function n_sqrt_test(bytes4 n_inv_sqrt, bytes4 n_bytes) public pure returns (bool) {
-        uint n_inv_sqrt_256 = uint256(uint32(n_inv_sqrt));
-        uint n = uint(uint32(n_bytes));
+    function n_sqrt_test(bytes8 n_inv_sqrt, bytes8 n_bytes) public pure returns (bool) {
+        uint n_inv_sqrt_256 = uint256(uint64(n_inv_sqrt));
+        uint n = uint(uint64(n_bytes));
         uint n_inv_sqrt_test = n_inv_sqrt_256 * n_inv_sqrt_256 * n >> 2 * fraction_bits;
         uint one_256 = 1 << fraction_bits;
         uint error = 2 * n_inv_sqrt_256 + 1;
         return one_256 - error <= n_inv_sqrt_test && n_inv_sqrt_test <= one_256 + error;
     }
 
-    function n1_check(bytes4 n_bytes, bytes4 n1_inv) public pure returns (bool) {
-        uint n = uint(uint32(n_bytes));
+    function n1_check(bytes8 n_bytes, bytes8 n1_inv) public pure returns (bool) {
+        uint n = uint(uint64(n_bytes));
         uint n1 = n - 1;
-        uint n1_inv_256 = uint256(uint32(n1_inv));
+        uint n1_inv_256 = uint256(uint64(n1_inv));
         uint error = 2 * n1_inv_256 + 1;
         uint n1_inv_test = n1_inv_256 * n1 >> fraction_bits;
         uint one_256 = 1 << fraction_bits;
         return one_256 - error <= n1_inv_test && n1_inv_test <= one_256 + error;
     }
 
-    function s2_check(uint256 claimed_s, bytes4 new_s2) public pure returns (bool) {
-        uint s_test = uint256(uint32(claimed_s));
-        uint s2 = uint256(uint32(new_s2));
+    function s2_check(uint256 claimed_s, bytes8 new_s2) public pure returns (bool) {
+        uint s_test = uint256(uint64(claimed_s));
+        uint s2 = uint256(uint64(new_s2));
         uint s2_test = s_test * s_test >> fraction_bits;
         uint error = 2 * s_test + 1;
         return s2_test < s2 + error && s2_test > s2 - error;
     }
 
-    function getVolatility() external view returns (uint256) {
-        return rv;
+ function calculateFee(uint256 volume, uint256 volatility, uint256 price) public view returns (uint24) {
+    // console.log(volume);
+    // console.log(volatility);
+    // console.log(price);
+
+    // Normalize inputs
+    uint256 maxVolume = 1e18; // max value for volume
+    uint256 maxVolatility = 1e9; // max value for volatility
+    uint256 maxPrice = 1e10; //  max value for price
+
+    uint256 normalizedVolume = volume * 1e18 / maxVolume;
+    uint256 normalizedVolatility = volatility * 1e18 / maxVolatility;
+    uint256 normalizedPrice = price * 1e18 / maxPrice;
+
+    uint256 value = (normalizedVolume + normalizedVolatility + normalizedPrice) / 3;
+
+    // Scale combined value to fee range
+    uint256 minFee = 1000;
+    //TODO: should the max fee exceed some limit?
+    uint256 maxFee = 100000;
+
+    uint256 fee = minFee + (value * (maxFee - minFee) / 1e18);
+
+    return uint24(fee);
     }
-    
-    function getPrice() external view returns (uint256) {
-        return price;
+
+    function getFee(bytes calldata data) external view returns (uint24){
```

Comment:
Hi Seve, we were discussing it in the uniswap v4 slack chat, deciding between either we change the name of the oracle contract to only SnarkBasedRvOracle (instead of SnarkBasedFeeOracle) or makes it a pure fee oracle. Sorry that we did not get your input on this! We will revert it back before redeploy!